parser_library/src/processing/context_manager.cpp

```diff
@@ -302,23 +163,8 @@ bool context_manager::test_symbol_for_read(
 
 void context_manager::collect_diags() const {}
 
-context::macro_data_ptr context_manager::create_macro_data(const semantics::concat_chain& chain,
-    const std::function<std::string(const semantics::concat_chain& chain)>& to_string) const
+context_manager::~context_manager()
 {
-    context_manager mngr(hlasm_ctx);
-
-    if (chain.size() == 0)
-        return std::make_unique<context::macro_param_data_dummy>();
-    else if (chain.size() > 1 || (chain.size() == 1 && chain[0]->type != semantics::concat_type::SUB))
-        return std::make_unique<context::macro_param_data_single>(to_string(chain));
-
-    const auto& inner_chains = chain[0]->access_sub()->list;
-
-    std::vector<context::macro_data_ptr> sublist;
-
-    for (auto& inner_chain : inner_chains)
-    {
-        sublist.push_back(create_macro_data(inner_chain, to_string));
-    }
-    return std::make_unique<context::macro_param_data_composite>(std::move(sublist));
+    if (eval_ctx_)
+        eval_ctx_->collect_diags_from_child(*this);
```

Comment:
you are in destructor, so if this throws (e.g. `std::bad_alloc`) it will terminate the whole program.